src/io.c

```diff
@@ -1910,23 +1910,26 @@ static Obj FuncPRINT_FORMATTING_ERROUT(Obj self)
     return IO()->PrintFormattingForErrout ? True : False;
 }
 
-static Obj FuncSET_PRINT_FORMATTING_CURRENT(Obj self, Obj val)
+/****************************************************************************
+**
+*F  FuncCALL_WITH_FORMATTING_STATUS( <status>, <func>, <args> )
+**
+**  Temporarily set the formatting status of the active output stream to
+**  <status>, then call the function <func> with the arguments in <args>.
+*/
+static Obj FuncCALL_WITH_FORMATTING_STATUS(Obj self, Obj status, Obj func, Obj args)
 {
-    TypOutputFile * output = IO()->Output;
-    if (!output)
-        ErrorMayQuit("SET_PRINT_FORMATTING_CURRENT called while no output is open", 0, 0);
-    output->format = (val != False);
-    if (output->stream)
-        CALL_2ARGS(SetPrintFormattingStatus, output->stream, val);
-    return 0;
-}
+    RequireTrueOrFalse(SELF_NAME, status);
+    RequireSmallList(SELF_NAME, args);
 
-static Obj FuncPRINT_FORMATTING_CURRENT(Obj self)
-{
     TypOutputFile * output = IO()->Output;
     if (!output)
-        ErrorMayQuit("PRINT_FORMATTING_CURRENT called while no output is open", 0, 0);
-    return output->format ? True : False;
+        ErrorMayQuit("CALL_WITH_FORMATTING_STATUS called while no output is open", 0, 0);
+    BOOL old = output->format;
+    output->format = (status != False);
+    Obj result = CallFuncList(func, args);
```

Comment:
I was a bit worried about what happens if this enters the break loop, and the user exits, which leads to broken formatting (this isn't new to this PR, but maybe worth fixing anyway).

Maybe replace the line `Obj result = CallFuncList(func,args);` with something like:

    Obj result = 0;
    GAP_TRY
    {
      result = CallFuncList(func, args);
    }
    GAP_CATCH
    {
      output->format = old;
      GAP_THROW();
    }

Reply:
Good point, done.

I also added a similar tweak to `CALL_WITH_STREAM` to restore the active output if there is an error. Of course many other kernel functions still don't do such a thing